Functions/FirebaseFunctions/FIRFunctions.m

```diff
@@ -75,16 +101,23 @@ + (instancetype)functionsForApp:(FIRApp *)app region:(NSString *)region {
 }
 
 - (instancetype)initWithApp:(FIRApp *)app region:(NSString *)region {
+  return [self initWithApp:app region:region auth:FIR_COMPONENT(FIRAuthInterop, _app.container)];
+}
+
+- (instancetype)initWithApp:(FIRApp *)app
+                     region:(NSString *)region
+                       auth:(nullable id<FIRAuthInterop>)auth {
   self = [super init];
   if (self) {
     if (!region) {
       FUNThrowInvalidArgument(@"FIRFunctions region cannot be nil.");
     }
     _fetcherService = [[GTMSessionFetcherService alloc] init];
     _app = app;
+    _auth = auth;
     _region = [region copy];
     _serializer = [[FUNSerializer alloc] init];
-    _contextProvider = [[FUNContextProvider alloc] initWithApp:app];
+    _contextProvider = [[FUNContextProvider alloc] initWithAuth:_auth];
```

Comment:
Nit: Along with the above, we can just use `auth` instead of assigning it to an instance var since it's not used outside of this scope.

Reply:
Done.